src/ExpressionEvaluator/Core/Test/ExpressionCompiler/MockSymUnmanaged.cs

```diff
@@ -331,6 +337,43 @@ int ISymUnmanagedNamespace.GetVariables(int cVars, out int pcVars, ISymUnmanaged
         }
     }
 
+    internal delegate int GetSignatureDelegate(int bufferLength, out int count, byte[] signature);
+
+    internal sealed class MockSymUnmanagedConstant : ISymUnmanagedConstant
+    {
+        private readonly string _name;
+        private readonly object _value;
+        private readonly GetSignatureDelegate _getSignature;
+
+        public MockSymUnmanagedConstant(string name, object value, GetSignatureDelegate getSignature)
+        {
+            _name = name;
+            _value = value;
+            _getSignature = getSignature;
+        }
+
+        int ISymUnmanagedConstant.GetName(int bufferLength, out int count, char[] name)
+        {
+            count = _name.Length + 1; // + 1 for null terminator
+            for (int i = 0; i < bufferLength - 1; i++)
```

Comment:
I was content to let it throw or not leave a null terminator (which is checked for elsewhere), but now that you point it out, I might as well explicitly assert ```bufferLength == count``` here.